pkg/cmd/server/origin/controller/build.go

```diff
@@ -25,20 +23,10 @@ type BuildControllerConfig struct {
 
 // RunController starts the build sync loop for builds and buildConfig processing.
 func (c *BuildControllerConfig) RunController(ctx ControllerContext) (bool, error) {
-	kubePluginInitializer := kubeadmission.NewPluginInitializer(
-		ctx.ClientBuilder.KubeInternalClientOrDie(bootstrappolicy.InfraBuildControllerServiceAccountName),
-		ctx.InternalKubeInformers,
-		nil, // api authorizer, only used by PSP
-		nil, // cloud config
-		nil, // quota registry
-	)
-	originPluginInitializer := &oadmission.PluginInitializer{
-		SecurityInformers: ctx.SecurityInformers,
-	}
-	allPluginInitializers := admission.PluginInitializers([]admission.PluginInitializer{kubePluginInitializer, originPluginInitializer})
-
-	admissionControl, err := admission.InitPlugin("SecurityContextConstraint", nil, allPluginInitializers)
-	if err != nil {
+	sccAdmission := sccadmission.NewConstraint()
+	sccAdmission.SetSecurityInformers(ctx.SecurityInformers)
+	sccAdmission.SetInternalKubeClientSet(ctx.ClientBuilder.KubeInternalClientOrDie(bootstrappolicy.InfraBuildControllerServiceAccountName))
+	if err := sccAdmission.Validate(); err != nil {
```

Comment:
> looks like this resolves #14909 ?

No, this just makes the hack fail closer to compile time instead of a random failure at runtime.